engine/src/paragrafattr.cpp

```diff
@@ -100,11 +100,30 @@ void MCParagraph::fetchattrs(MCArrayRef src)
     t_color . name = nil;
 
     // AL-2014-09-30: [[ Bug 13559 ]] Don't explicitly set any paragraph styles that aren't in the array (even to empty)
-    if (ctxt . CopyElementAsInteger(src, MCNAME("textAlign"), false, t_intenum_value))
-        SetTextAlign(ctxt, &t_intenum_value);
-
-    if (ctxt . CopyElementAsInteger(src, MCNAME("listStyle"), false, t_intenum_value))
-        SetListStyle(ctxt, t_intenum_value);
+	// PM-2015-10-14: [[ Bug 16210 ]] Get the stringref for textAlign from the src array and then map it to the correct
+	// intenum_t, to be used in SetTextAlign()
+	if (ctxt . CopyElementAsString(src, MCNAME("textAlign"), false, t_stringref_value))
```

Comment:
This seems perfect for a MCExecContext::CopyElementAsEnum(array, keyname, casesensitive, enumtypeinfo, r_intenum) abstraction - which can be used again below too.